src/renderer/common.ts

```diff
@@ -26,33 +34,40 @@ export enum MeshPrimitive {
   Lines,
 }
 
-export interface Buffer {
-  bufferData: ArrayBuffer
-  glBuffer: WebGLBuffer
+export interface Buffer<T> {
+  buffer: T
```

Comment:
This is an artifact of the previous implementation, but this refactor now exposes the fact that using a typed buffer (Float32Array, etc.) here might be redundant in the presence of `glType`. There's probably another refactor we can do to either turn `buffer` into an `ArrayBuffer`, or get rid of `glType`.